modules/kvformat/linux-audit-scanner.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2015 BalaBit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "kv-scanner.h"
+#include "misc.h"
+#include "utf8utils.h"
+
+#include <string.h>
+#include <ctype.h>
+
+const gchar *hexcoded_fields[] =
+{
+  "name",
+  "proctitle",
+  "path",
+  "dir",
+  "comm",
+  "ocomm",
+  "data",
+  "old",
+  "new",
+  NULL
+};
+
+static gint
+_decode_xdigit(gchar xdigit)
+{
+  if (xdigit >= '0' && xdigit <= '9')
+    return xdigit - '0';
+  xdigit = toupper(xdigit);
+  if (xdigit >= 'A' && xdigit <= 'F')
+    return xdigit - 'A' + 10;
+  return -1;
+}
+
+static gint
+_decode_xbyte(gchar xdigit1, gchar xdigit2)
+{
+  gint nibble_hi, nibble_lo;
+
+  nibble_hi = _decode_xdigit(xdigit1);
+  nibble_lo = _decode_xdigit(xdigit2);
+  if (nibble_hi < 0 || nibble_lo < 0)
+    return -1;
+  return (nibble_hi << 4) + nibble_lo;
+}
+
+static gboolean
+_parse_linux_audit_hexstring(GString *decoded_value, const gchar *value, gsize len)
+{
+  gint src;
+  gboolean kernel_would_have_encoded_this_as_hex = FALSE;
+
+  for (src = 0; src < len; src += 2)
+    {
+      gint v;
+
+      v = _decode_xbyte(value[src], value[src + 1]);
+
+      if (v < 0)
+        return FALSE;
+      if (v < 0x21 || v >= 0x7e || v == '"' || v == 0)
```

Comment:
The `0x7e` character is the `~` character. Why do you need to quote it (or just the `>=` is wrong)?